### PR TITLE
feat: add stripe webhook

### PR DESCRIPTION
**Pull request by altaywtf**
feat: add  to env

feat: add more params to stripe client

feat: add stripe webhook

feat: update parser

### CODE REVIEW

**Comment by altaywtf** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #56** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/56?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #57** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/57?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #59** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/59?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #61** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/61?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #62** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/62?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #60** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/60?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #63** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/63?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #65** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/65?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #66** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/66?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #69** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/69?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                      * **PR #67** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/67?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                        * **PR #70** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/70?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                          * **PR #73** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/73?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                            * **PR #74** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/74?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                              * **PR #75** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/75?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                * **PR #76** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/76?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
                          * **PR #72** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/72?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #64** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/64?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #58** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/58?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/experiment-station/beecast/76?utm_source=stack-comment).

File: lib/services/stripe/order.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  const updateAccountQuery = await supabase
+    .from('account')
+    .update({
+      ai_credit:
+        (currentAccountCreditsQuery.data.ai_credit ?? 0) +
+        validatedSession.data.line_items.data[0].price.transform_quantity
+          .divide_by,
+    })
```
Comment (okanisildar) [inline lib/services/stripe/order.ts:53]: nitpick: we can move this calculation outside of the supabase query to make it more readible.

**Review by okanisildar — APPROVED** [review]

File: lib/services/stripe/order.ts (hunk 1 above)
Comment (altaywtf) [reply lib/services/stripe/order.ts:53]: done it in the #78

**Comment by altaywtf** [comment]
### Merge activity

* **Dec 16, 7:08 AM**: [Graphite](https://app.graphite.dev/github/pr/experiment-station/beecast/76) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**